File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        string l[] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        set<string> res;
        for(int i=0;i<words.size();i++){
            string s = "";
            for(int j=0;j<words[i].length();j++)
                s+=l[words[i][j]-'a'];
            res.insert(s);
            }
        return res.size();
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        l = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        res = set()
        for i in words:
            s = ""
            for j in i:
                s+=l[ord(j)-ord('a')]
            res.add(s)
        return len(res)